EarlGrey/Action/GREYMultiFingerSwipeAction.m

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#import "GREYMultiFingerSwipeAction.h"
+
+#import "Action/GREYPathGestureUtils.h"
+#import "Additions/NSError+GREYAdditions.h"
+#import "Additions/NSObject+GREYAdditions.h"
+#import "Additions/NSString+GREYAdditions.h"
+#import "Assertion/GREYAssertionDefines.h"
+#import "Assertion/GREYAssertions+Internal.h"
+#import "Common/GREYError.h"
+#import "Event/GREYSyntheticEvents.h"
+#import "Matcher/GREYAllOf.h"
+#import "Matcher/GREYMatcher.h"
+#import "Matcher/GREYMatchers.h"
+#import "Matcher/GREYNot.h"
+
+@implementation GREYMultiFingerSwipeAction {
+  /**
+   *  The direction in which the content must be scrolled.
+   */
+  GREYDirection _direction;
+  /**
+   *  The duration within which the swipe action must be complete.
+   */
+  CFTimeInterval _duration;
+  /**
+   *  Start point for the swipe specified as a percentage 
+   *  of the swipped element's accessibility frame.
+   */
+  CGPoint _startPercents;
+  /**
+   *  Number of parallel swipes
+   */
+  NSUInteger _numberOfFingers;
+}
+
+- (instancetype)initWithDirection:(GREYDirection)direction
```

Comment:
Sorry I missed those. That was all introduced when the designated initializer was marked in the header file. I'll address those